aim/db/migration/alembic_migrations/versions/5d975a5c2d60_vmm_injected_objects.py

```diff
@@ -0,0 +1,228 @@
+# Copyright (c) 2016 Cisco Systems
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""Tables for VMM injected objects
+
+Revision ID: 5d975a5c2d60
+Revises: 32e4c4d73dfc
+Create Date: 2017-04-17 16:07:35.571565
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '5d975a5c2d60'
+down_revision = 'aabce110030f'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.create_table(
+        'aim_vmm_controllers',
+        sa.Column('aim_id', sa.Integer, autoincrement=True),
+        sa.Column('domain_type', sa.String(64), nullable=False),
+        sa.Column('domain_name', sa.String(64), nullable=False),
+        sa.Column('name', sa.String(64), nullable=False),
+        sa.Column('display_name', sa.String(256), nullable=False, default=''),
+        sa.Column('monitored', sa.Boolean, nullable=False, default=False),
+        sa.Column('scope', sa.Enum('unmanaged', 'vm', 'iaas', 'network',
```

Comment:
I don't think DB migration is a problem. Generally speaking, whether the backend store should be more restrictive in accepting arbitrary values for enumerations is a topic worth discussing. It is however out of scope of this change because we should fix all enumerations instead of starting a new pattern for these objects.